### PR TITLE
Simplify normalization of field/var initialization

### PR DESCRIPTION
**Pull request by benharsh**
This PR collapses and merges a bunch of conditionals during normalization that handled field/variable initialization depending on the expression's result type. Since we can't reliably know the type at this stage of compilation, this PR pushes such handling towards resolution.

Also fixes some unknown/untested bugs involving usage of qualified "MOD.TYPE" expressions in new-exprs.

Testing:
- [x] local + futures
- [x] gasnet + futures
- [x] memleaks

### CODE REVIEW

File: test/compflags/bradc/mungeUserIdents/testmunge-extern.chpl [hunk 1]
```diff
@@ -4,12 +4,16 @@ extern record bar {
...
 
-extern var baz: bar;
-var b: bar;
-
-foo();
-writeln(baz);
-writeln(baz.foo);
-b.foo = 1.2;
-writeln(b);
+// Use 'main' to ensure that 'bar' is present in the generatec C for this
```
Comment (lydia-duncan) [inline test/compflags/bradc/mungeUserIdents/testmunge-extern.chpl]: Typo on "generatec"

File: test/compflags/bradc/mungeUserIdents/testmunge-extern.chpl [hunk 2]
```diff
@@ -4,12 +4,16 @@ extern record bar {
...
+// module. If it's global, it will be in the header file where this test's
+// prediff will not look.
```
Comment (lydia-duncan) [inline test/compflags/bradc/mungeUserIdents/testmunge-extern.chpl]: I would argue that implies we should add a test case that checks the header file

Comment (benharsh) [reply test/compflags/bradc/mungeUserIdents/testmunge-extern.chpl]: Could that be part of a separate PR (which I will start right away) just to keep this branch cleaner?

Comment (lydia-duncan) [reply test/compflags/bradc/mungeUserIdents/testmunge-extern.chpl]: I'm okay with that

**Pull request merged**